visualization/app/codeCharta/ui/edgeChooser/edgeChooser.po.ts

```diff
@@ -8,4 +10,28 @@ export class EdgeChooserPageObject {
 		await page.waitForSelector(".edge-metric")
 		return await page.$$eval(".edge-metric", metrics => metrics.map(x => x.textContent))
 	}
+
+	public async selectEdgeMetric(metric: string) {
+		await this.open()
+		await expect(page).toClick(`#edge-metric-${metric}`, { timeout: 3000 })
+	}
+
+	public async isEdgeCountAvailable() {
+		const innerText = await this.getEdgeCountInnerText()
+
+		// Check if the text contains a number.
+		return /\d/.test(innerText)
+	}
+
+	public async getAmountOfEdges(): Promise<EdgeMetricCount> {
+		const innerText = await this.getEdgeCountInnerText()
+		const edgeCount = innerText.split("/")
+
+		return { incoming: parseInt(edgeCount[0]), outgoing: parseInt(edgeCount[1]) }
```

Comment:
`parseInt` and `parseFloat` are strong anti-patterns. They parse faulty data in weird ways (e.g., `parseInt('1aeas9jd8')` === 1???). Instead, always use `Number()` for floats. If you want to guarantee it's an integer, use `Math.floor()` or `Math.ceil()` or `Number.isInteger(Number())` or `| 0` for SMIs (small integers = 2 ** 31 - 1) or `>>> 0` for 2 ** 32 .